version: '3.8'

services:
  db:
    image: postgres:latest
    environment:
      POSTGRES_USER: ${DB_USER_POS}
      POSTGRES_PASSWORD: ${DB_PASSWORD_POS}
      POSTGRES_DB: ${DB_NAME_POS}
    volumes:
      - postgres_data:/var/lib/postgresql/data
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U ${DB_USER_POS}"]
      interval: 15s
      timeout: 5s
      retries: 3
    command: ["docker-entrypoint.sh", "postgres"]

  todoapp:
    build: .
    volumes:
      - .:/app
    environment:
      SECRET_KEY: ${SECRET_KEY}
    ports:
      - "8888:8888"
    command: sh -c "python3 manage.py runserver 0.0.0.0:8888"
    depends_on:
      db:
        condition: service_healthy

volumes:
  postgres_data:
